lab2.js

```diff
@@ -127,24 +190,80 @@ function lastLetterSort(stringArray) {
     // this byLastLetter function is a "compare function"
     // And check out the "comparing strings" section  here:
     // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String
+    var aLength = a.length;
+    var bLength = b.length;
+
+    if (a.charAt((aLength - 1)) < b.charAt((bLength - 1))) {
+      return -1;
+    }
+    if (a.charAt((aLength - 1)) > b.charAt((bLength - 1))) {
+      return 1;
+    }
+    return 0;
   }
-  stringArray.sort(byLastLetter);
+
+  return stringArray.sort(byLastLetter);
 }
 
+//lastLetterSort(myStringArray);
+//lastLetterSort(myString2Array);
+
 function sumArray(numberArray) {
   var sum = 0;
   // TODO: implement me using forEach
+  function sumOfElements(element) {
+    sum += element;
+  }
+
+  numberArray.forEach(sumOfElements);
+
   return sum;
 }
 
+//console.log(sumArray(myNumberArray1));
+//console.log(sumArray(myNumberArray2));
+//console.log(sumArray(myNumberArray3));
+
 function sumSort(arrayOfArrays) {
-  arrayOfArrays.sort(function(item) {
+  arrayOfArrays.sort(function(item1, item2) {
     // TODO: implement me using sumArray
     //  order the arrays based on the sum of the numbers
     //  inside each array
+    //console.log(item1 +"   "+ item2)
+    if (sumArray(item1) <  sumArray(item2)) {
+      return -1;
+    }
+    if (sumArray(item1) > sumArray(item2)) {
+      return 1;
+    }
+    return 0;
   });
+  return arrayOfArrays;
 }
 
+//console.log("1: " + sumSort(numList1Array));
+//console.log("2: " + sumSort(numList2Array));
+
+//Not sure this is the right way to compare arrays but only way I knew I could get it to work.
```

Comment:
this works just fine. You just have to pick something and go with it, since you can't compare them directly.